contracts/Core/AssetManager.sol

```diff
@@ -0,0 +1,300 @@
+            assetTypes.Job
+        );
+    }
+
+    function fulfillAsset(
+        uint256 id, 
+        uint256 value   
+    ) 
+        external 
+        onlyRole(parameters.getAssetConfirmerHash())
+    {
+        uint256 epoch = parameters.getEpoch();
+        if(jobs[id].assetType == uint256(assetTypes.Job)){
+
+            Structs.Job storage job = jobs[id];
+
+            if (!job.repeat) {
+                job.fulfilled = true;
+            }
+
+            job.result = value;
+            emit JobReported(
+                job.id,
+                value,
+                epoch,
+                job.url,
+                job.selector,
+                job.name,
+                job.repeat,
+                job.creator,
+                job.credit,
+                job.fulfilled,
+                block.timestamp
+            );
+        }
+        else if(collections[id].assetType==uint256(assetTypes.Collection)){
+
+            Structs.Collection storage collection = collections[id];
+
+            collection.result = value;
+
+            emit CollectionReported(
+                collection.id,
+                value,
+                epoch,
+                collection.name,
+                collection.aggregationMethod,
+                collection.jobIDs,
+                collection.creator,
+                collection.credit,
+                block.timestamp
+            );
+        }
+    }
+
+    function createCollection(
+        string calldata name, 
+        uint256[] memory jobIDs,
+        uint32 aggregationMethod
+    ) external payable 
+    {
+        require(aggregationMethod > 0 && aggregationMethod < parameters.aggregationRange(),"Aggregation range out of bounds");
+        require(jobIDs.length > 1,"Number of jobIDs low to create collection");
+
+        numAssets = numAssets + 1;
+        uint256 epoch = parameters.getEpoch();
+        collections[numAssets].id = numAssets; 
+        collections[numAssets].name = name; 
+        collections[numAssets].aggregationMethod = aggregationMethod; 
+        collections[numAssets].epoch = epoch;
+        collections[numAssets].creator = msg.sender;
+        collections[numAssets].credit = msg.value;
+        for(uint256 i = 0; i < jobIDs.length; i++){
+            require(jobs[jobIDs[i]].assetType==uint256(assetTypes.Job),"Job ID not present");
+            require(!collections[numAssets].jobIDExist[jobIDs[i]],"Duplicate JobIDs sent");
+            collections[numAssets].jobIDs.push(jobIDs[i]);
+            collections[numAssets].jobIDExist[jobIDs[i]] = true;
+        }
+        collections[numAssets].assetType = uint256(assetTypes.Collection);
+
+        emit CollectionCreated(
+            numAssets, 
+            epoch,
+            name, 
+            aggregationMethod,
+            jobIDs,
+            msg.sender, 
+            msg.value, 
+            block.timestamp, 
+            assetTypes.Collection
+        );
+    }
+
+    function addJobToCollection(
```

Comment:
is there a remove JobFromCollection() method? That is also required. you can create a separate issue and PR for that @SamAg19

Reply:
Yes, this method can be part of the solution that will be introduced for issue #86